packages/dds/map/src/mapKernel.ts

```diff
@@ -440,18 +455,61 @@ export class MapKernel {
         return false;
     }
 
+    /**
+     * Rollback a local op
+     * @param op - The operation to rollback
+     * @param localOpMetadata - The local metadata associated with the op.
+     */
+    public rollback(op: any, localOpMetadata: unknown) {
+        if (op.type === "clear") {
+            const metadata = localOpMetadata as IMapClearLocalOpMetadata;
+            if (metadata.previousMap === undefined) {
+                throw new Error("Cannot rollback without previous map");
+            }
+            metadata.previousMap.forEach((localValue, key) => {
+                this.data.set(key, localValue);
```

Comment:
what does the clear event look like? we probably want some event here. i kind of think we just call setCore on all keys so they all re-appear